server/server.go

```diff
@@ -104,6 +104,10 @@ type Config struct {
 	HTTPServerReadHeaderTimeout   time.Duration `yaml:"http_server_read_header_timeout"`
 	HTTPServerWriteTimeout        time.Duration `yaml:"http_server_write_timeout"`
 	HTTPServerIdleTimeout         time.Duration `yaml:"http_server_idle_timeout"`
+	// HTTPServerWriteTimeout has some misleading behavior. Consider using the below fields to
+	// enable a timeout handler that will write a response after the timeout. See tests for details.
+	HTTPServerUseTimeoutHandler     bool   `yaml:"-"`
```

Comment:
can we maybe include that's misleading? tests aren't available when the library is venodred. Perhaps something along the lines of 

```
Use http.TimeoutHandler instead of http.Server.WriteTimeout (See also middleware.TimeoutMiddleware).
Using the WriteTimeout means that the request's context is not cancelled 
immediatelly after hitting the timeout and that the TCP connection is closed.
When using the TimeoutMiddleware, HTTP 503 is instead gracefully returned.
When using the TimeoutMiddleware any later handler will be bound by its timeout,
whereas if you're relying on the server write timeout,
any handler in the chain can modify the write deadline via a ResponseController.
```

Reply:
`http.TimeoutHandler` also has one serious drawback -- it buffers entire result in memory until handler finishes, and disallows streaming.

Reply:
moved all comments to middleware and removed Server changes. also added @pstibrany's concern about response buffering.